fastai/train.py

```diff
@@ -37,7 +37,7 @@ def lr_find(learn:Learner, start_lr:Floats=1e-7, end_lr:Floats=10, num_it:int=10
     end_lr = learn.lr_range(end_lr)
     end_lr = np.array(end_lr) if is_listy(end_lr) else end_lr
     cb = LRFinder(learn, start_lr, end_lr, num_it, stop_div)
-    epochs = int(np.ceil(num_it/len(learn.data.train_dl)))
+    epochs = int(np.ceil(num_it/len(learn.data.train_dl))) * (num_distrib() or 1)
```

Comment:
Why do you need to multiply by `num_distrib` here?

Reply:
Hi Sylvain, thank you for the question.  This line compensates for inadequate number of epochs when attempting to `lr_find()` in distributed training mode.

As `LRFinder` steps through a range of learning rate only inside `on_batch_end()`, the more the # of batches it processes, the more learning rates it tries.  If the **total # of batches processed** is not large enough, `lr_find()` may finish without reaching a learning rate at which the loss started to explode.

In distributed training with multiple GPUs: the # of batches per epoch is scaled down, but the above line 40 **does not** scale up the **# of epochs** enough.    The change I propose compensate for that, by scaling up **# of epochs by # of GPUs**, so that the **total # of batches** will be big enough for `lr_find()`.  Early-stopping still works with this change.

Does this make sense?

------- evidence from experiments below -------
See the three graphs below for comparison of a successful `lr_find` in single GPU mode, and an unsuccessful one in distributed training, using the same `bs`, then lastly with the change, a successful `lr_find()` that has up to 150 batches to find the exploding-loss learning rate.

E.g. **lesson3-camvid** in course v3, dl1, single GPU (GTX 1080ti), and `bs=8`.

1. Graph 1 --- the number of batches per epoch is determined  to be **75**, and it will find the loss-explosion after at around batch # ~68 - 72-ish, the **range of learning rate** covered is `1e-06` -- `1e-02`:

2. Graph 2 ---   Scaling to use 3 GPUs in distributed training, `lr_find()` would run each epoch with only **75/3 = 25** batches, but the original code only sets "epochs" to 2.  As such, `LRFinder.on_batch_end()`only gets to process **25 x 2 = 50** batches, and the **range of learning rates** searched ended before`1e-03`, much shorter than case 1 above, without finding the learning rate at which loss exploded.

3. Graph 3 --- Then with this PR, in distributed training, `epochs * (num_distrib() or 1)` changes it from 2 to 6, and that gives a max # of batches of `6 * 25 = 150`.  See the range of learning rate covered is good again.

**Graph 1: single GPU, bs=8, max # of batches to try = 75, lr covered: 1e-06 -- 1e-02**
![bs=8, non-DDP](https://user-images.githubusercontent.com/19887230/74327625-5dcab480-4d41-11ea-8857-5c9371970191.png)

**Graph 2: three GPUs, bs=8, max # of batches to try = 25 x 2 = 50 only, , lr covered: 1e-06 -- 7e-03***
![bs=8 DDP bn=25, epoch = 2](https://user-images.githubusercontent.com/19887230/74327557-41c71300-4d41-11ea-8283-2c5c815ae4d7.png)


**Graph 3: three GPUs, bs=8, max # of batches to try = 25 x 6 = 150, lr covered: 1e-06 -- 1e-02**
![Screen Shot 2020-02-12 at 2 58 34 AM](https://user-images.githubusercontent.com/19887230/74328896-98355100-4d43-11ea-85c4-8e2bb53525b7.png)

Reply:
Got it, thanks for explaining it to me!